temboardui/plugins/monitoring/static/js/temboard.monitoring.js

```diff
@@ -41,8 +39,25 @@ function new_graph(id, title, api, api_url, options, start_date, end_date)
         return m.toDate().getTime();
       },
       drawCallback: function(g, is_initial) {
-        add_visibility_cb(id, g, is_initial);
-        add_export_button_callback(id, g, is_initial, title);
+        if (g.numRows() == 0)
+        {
+          $('#info'+id).html('<center><i>No data available</i></center>');
+          $('#legend'+id).css('visibility', 'hidden');
+          $('#legend'+id).css('display', 'none');
+          $('#chart'+id).css('visibility', 'hidden');
+          $('#chart'+id).css('display', 'none');
+          $('#visibility'+id).css('visibility', 'hidden');
+          $('#visibility'+id).css('display', 'none');
+        } else {
+          add_visibility_cb(id, g, is_initial);
+          $('#info'+id).html('');
+          $('#legend'+id).css('visibility', 'visible');
+          $('#legend'+id).css('display', 'block');
+          $('#chart'+id).css('visibility', 'visible');
+          $('#chart'+id).css('display', 'block');
+          $('#visibility'+id).css('visibility', 'visible');
+          $('#visibility'+id).css('display', 'block');
+        }
```

Comment:
Is it required to change both `visibility` and `display`?
Could you try using `Jquery` `.show()` and `.hide()` and see what happens?

Reply:
Thank you ! `.show()` and `.hide()` do the job.